### PR TITLE
backend/enhancement-redis changes

### PR DESCRIPTION
**Pull request by RitikaHanish**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
added fraud configs in non critical redis .

PR 
https://github.com/nammayatri/beckn-gateway/pull/45

### CODE REVIEW

File: Backend/lib/scheduler/src/Lib/Scheduler/App.hs [hunk 1]
```diff
@@ -47,6 +47,11 @@ runSchedulerService SchedulerConfig {..} handle_ = do
   esqDBEnv <- prepareEsqDBEnv esqDBCfg loggerEnv
   coreMetrics <- Metrics.registerCoreMetricsContainer
   hedisEnv <- connectHedis hedisCfg (\k -> hedisPrefix <> ":" <> k)
+  hedisNonCriticalEnv <- connectHedis hedisCfg (\k -> hedisPrefix <> ":" <> k)
```
Comment (NikithShetty) [inline Backend/lib/scheduler/src/Lib/Scheduler/App.hs:50]: This should be `hedisNonCriticalCfg`?

File: Backend/lib/scheduler/src/Lib/Scheduler/App.hs [hunk 2]
```diff
@@ -47,6 +47,11 @@ runSchedulerService SchedulerConfig {..} handle_ = do
...
+  hedisNonCriticalClusterEnv <-
+    if cutOffHedisCluster
+      then pure hedisNonCriticalEnv
+      else connectHedisCluster hedisClusterCfg (\k -> hedisPrefix <> ":" <> k)
```
Comment (NikithShetty) [inline Backend/lib/scheduler/src/Lib/Scheduler/App.hs:54]: This should be `hedisNonCriticalClusterCfg`?

File: Backend/app/rider-platform/rider-app/Main/src/Tools/SignatureAuth.hs [hunk 3]
```diff
@@ -95,6 +95,8 @@ instance
     KnownSymbol header,
     FromJSON req,
     HasField "hedisEnv" r Redis.HedisEnv,
+    HasField "hedisNonCriticalEnv" r Redis.HedisEnv,
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Tools/SignatureAuth.hs:98]: Is it required here?
Signature auth is always in critical path.

File: Backend/app/dashboard/Lib/src/Environment.hs [hunk 4]
```diff
@@ -92,11 +96,17 @@ buildAppEnv authTokenCacheKeyPrefix AppCfg {..} = do
   esqDBReplicaEnv <- prepareEsqDBEnv esqDBReplicaCfg loggerEnv
   coreMetrics <- registerCoreMetricsContainer
   let modifierFunc = ("dashboard:" <>)
+  let nonCriticalModifierFunc = ("dashboard:non-critical:" <>)
```
Comment (NikithShetty) [inline Backend/app/dashboard/Lib/src/Environment.hs:99]: can you provide shorter key name? `d:n_c:`?
Trying to reduce key's memory footprint

File: Backend/app/rider-platform/rider-app/Main/src/Environment.hs [hunk 5]
```diff
@@ -168,11 +172,17 @@ buildAppEnv AppCfg {..} = do
   esqDBReplicaEnv <- prepareEsqDBEnv esqDBReplicaCfg loggerEnv
   kafkaProducerTools <- buildKafkaProducerTools kafkaProducerCfg
   kafkaEnvs <- buildBAPKafkaEnvs
+  let nonCriticalModifierFunc = ("app-backend:non-critical:" <>)
```
Comment (NikithShetty) [inline Backend/app/rider-platform/rider-app/Main/src/Environment.hs:175]: can you provide shorter key name? `ab:n_c:`?

**Review by NikithShetty — DISMISSED** [review]

**Review by piyushKumar-1 — APPROVED** [review]

**Pull request merged**